java/arcs/core/data/expression/Builders.kt

```diff
@@ -213,6 +213,31 @@ open class MapScope<V>(
   }
 
   override fun toString() = map.toString()
+
+  private val allKeys get() = map.keys + (parentScope?.map?.keys ?: emptySet())
```

Comment:
What's the type of this? I find it unclear.

Reply:
its a set